lib/Serialization/SerializeSIL.cpp

```diff
@@ -2469,6 +2491,54 @@ writeSILDefaultWitnessTable(const SILDefaultWitnessTable &wt) {
   }
 }
 
+// SWIFT_ENABLE_TENSORFLOW
+void SILSerializer::
+writeSILDifferentiabilityWitness(const SILDifferentiabilityWitness &dw) {
+  Mangle::ASTMangler mangler;
+  auto mangledKey = mangler.mangleSILDifferentiabilityWitnessKey(dw.getKey());
+  size_t nameLength = mangledKey.size();
+  char *stringStorage =
+      static_cast<char *>(StringTable.Allocate(nameLength, 1));
+  std::memcpy(stringStorage, mangledKey.data(), nameLength);
+  DifferentiabilityWitnessList[StringRef(stringStorage, nameLength)] =
+      NextDifferentiabilityWitnessID++;
+  DifferentiabilityWitnessOffset.push_back(Out.GetCurrentBitNo());
+
+  auto *original = dw.getOriginalFunction();
+  addReferencedSILFunction(original, /*DeclOnly*/ true);
+  IdentifierID jvpID = 0;
+  IdentifierID vjpID = 0;
+  if (auto *jvp = dw.getJVP()) {
+    addReferencedSILFunction(jvp, /*DeclOnly*/ true);
+    jvpID = S.addUniquedStringRef(jvp->getName());
+  }
+  if (auto *vjp = dw.getVJP()) {
+    addReferencedSILFunction(vjp, /*DeclOnly*/ true);
+    vjpID = S.addUniquedStringRef(vjp->getName());
+  }
+  SmallVector<unsigned, 8> parameterAndResultIndices(
+       dw.getParameterIndices()->begin(), dw.getParameterIndices()->end());
```

Comment:
```suggestion
      dw.getParameterIndices()->begin(), dw.getParameterIndices()->end());
```

Reply:
Done in https://github.com/apple/swift/pull/27487/commits/2073f863826fb305c1fc105a520d84de1e1463ec.